barretenberg/cpp/src/barretenberg/protogalaxy/presumcheck_prover.cpp

```diff
@@ -0,0 +1,135 @@
+        witness_commitments.calldata_read_counts = commitment_key->commit(instance->proving_key->calldata_read_counts);
+        transcript->send_to_verifier(domain_separator + commitment_labels.calldata,
+                                     instance->witness_commitments.calldata);
+        transcript->send_to_verifier(domain_separator + commitment_labels.calldata_read_counts,
+                                     instance->witness_commitments.calldata_read_counts);
+    }
+}
+
+/**
+ * @brief Compute sorted witness-table accumulator and commit to the resulting polynomials.
+ *
+ */
+template <IsUltraFlavor Flavor> void PreSumcheckProver<Flavor>::execute_sorted_list_accumulator_round()
+{
+    auto& witness_commitments = instance->witness_commitments;
+    const auto& commitment_labels = instance->commitment_labels;
+
+    auto eta = transcript->template get_challenge<typename Flavor::FF>(domain_separator + "eta");
+    instance->compute_sorted_accumulator_polynomials(eta);
+
+    // Commit to the sorted witness-table accumulator and the finalized (i.e. with memory records) fourth wire
+    // polynomial
+    witness_commitments.sorted_accum = commitment_key->commit(instance->prover_polynomials.sorted_accum);
+    witness_commitments.w_4 = commitment_key->commit(instance->prover_polynomials.w_4);
+
+    transcript->send_to_verifier(domain_separator + commitment_labels.sorted_accum, witness_commitments.sorted_accum);
+    transcript->send_to_verifier(domain_separator + commitment_labels.w_4, witness_commitments.w_4);
+}
+
+/**
+ * @brief Compute log derivative inverse polynomial and its commitment, if required
+ *
+ */
+template <IsUltraFlavor Flavor> void PreSumcheckProver<Flavor>::execute_log_derivative_inverse_round()
+{
+    const auto& commitment_labels = instance->commitment_labels;
+
+    auto [beta, gamma] =
+        transcript->template get_challenges<typename Flavor::FF>(domain_separator + "beta", domain_separator + "gamma");
+    instance->relation_parameters.beta = beta;
+    instance->relation_parameters.gamma = gamma;
+    if constexpr (IsGoblinFlavor<Flavor>) {
+        // Compute and commit to the logderivative inverse used in DataBus
+        instance->compute_logderivative_inverse(beta, gamma);
+        instance->witness_commitments.lookup_inverses =
+            commitment_key->commit(instance->prover_polynomials.lookup_inverses);
+        transcript->send_to_verifier(domain_separator + commitment_labels.lookup_inverses,
+                                     instance->witness_commitments.lookup_inverses);
+    }
+}
+
+/**
+ * @brief Compute permutation and lookup grand product polynomials and their commitments
+ *
+ */
+template <IsUltraFlavor Flavor> void PreSumcheckProver<Flavor>::execute_grand_product_computation_round()
+{
+    auto& witness_commitments = instance->witness_commitments;
+    const auto& commitment_labels = instance->commitment_labels;
```

Comment:
commitment_labels are static and come from flavor so they could become a field in `PreSumcheckProver` rather than extracting from instance every time

Reply:
It's just making a const reference of them, so it should be fine?

Reply:
actually, next PR will do this and remove it from instance at the same time.